### PR TITLE
chore: add script to automate release steps

### PR DESCRIPTION
**Pull request by timotheeg**
Release steps are repetitive and systematic, we should make our life easier by automating as much as we can, such as:

- [X] version upgrade
- [X] create the release branch
- [X] push to staging
- [X] create the PR by extracting the changelog of the upgrade
- [x] sort the changelog into section
- [x] fetch test plans from the PRs being released and add to PR body

### CODE REVIEW

File: scripts/release_prep.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+git fetch --all --tags
+git reset --hard
+git pull
+git checkout develop
+git reset --hard origin/develop
+
+short_hash=$(git rev-parse --short HEAD)
+temp_release_branch=temp_${short_hash}
+
+git chekcout -b ${temp_release_branch}
```
Comment (tshuli) [inline scripts/release_prep.sh]: typo!!!
```suggestion
git checkout -b ${temp_release_branch}
```

Comment (timotheeg) [reply scripts/release_prep.sh]: OMG Thanks! 

LOL, this is a typo I always do, so I have a local alias and it prevented me from see from seeing it 😅 

```
[alias]
	chekcout = checkout
```

**Comment by timotheeg** [comment]
Is good to go @karrui!

File: scripts/release_prep.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+set -x
+
+# pre-requisites: install github CLI
+# - github documentation: https://github.com/cli/cli#installation
+# - github is remote 'origin'
+# - PRs use test section LAST with heading "## Tests"
+# - ALL build and release PRs start with "build: "
+
+git fetch --all --tags
+git reset --hard
```
Comment (karrui) [inline scripts/release_prep.sh:12]: this line is gonna suck if we have changes we forgot about... should we remove this line and just let the script error if the user has unstaged changes

**Review by karrui — APPROVED** [review]
lgtm, we can clean up the script again before next use

**Pull request merged**